os/board/imxrt1050-evk/src/imxrt_boot.c

```diff
@@ -260,6 +265,10 @@ void board_initialize(void)
 
 	imxrt_spi_initialize();
 
+#ifdef CONFIG_WATCHDOG
+	imxrt_wdog_initialize(CONFIG_WATCHDOG_DEVPATH, IMXRT_WDOG1);
```

Comment:
Actually we provide up_wdginitialize as a common interface to initialize watchdog.
In this case, imxrt should inplmenet up_wdginitialize, and then up_wdginitialize will be called in up_initialize after timer initialized as a common feature, like devnull, timer, and so on.
But if you do not want follow this rule, please remove up_wdginitialize from watchdog.h

Reply:
I removed up_wdinitialize in watchdog.h
Some modules, like gpio, timer, and so on, can be used from another, but wdinitialize is only used for board specific code.
Thanks.